src/details/ArborX_DetailsBufferOptimization.hpp

```diff
@@ -221,79 +144,27 @@ void queryImpl(ExecutionSpace const &space, TreeTraversal const &tree_traversal,
   // pre-condition: offset and out are preallocated. If buffer_size > 0, offset
   // is pre-initialized
 
+  using MapType =
+      Kokkos::UnorderedMap<Kokkos::pair<int, int>,
+                           typename OutputView::value_type, ExecutionSpace>;
+  MapType unordered_map(1000000);
```

Comment:
If buffer is preallocated, a reasonable guess here would be the size of `out` (or last element of `offset`), multiplied by some factor (2x or 3x) to have the fill under certai n threshold. It also depends on `MapType`.